app/sockets.py

```diff
@@ -0,0 +1,153 @@
+from .__init__ import *
+
+
+from .MyFunctions import *
+# from .models import *
+from .schemas import *
+from .routes import *
+
+
+@socketio.on('join', namespace='/rooms_namespace')
+def on_join(data):
+
+    """
+    verify if attendence is success 
+    then get them join in that room name with otp added
+    now broadcast to that room
+    """
+    try:
+        data = json.loads(data) # now it's dict form, before it was string
+    except:
+        data = {}
+
+    status = data.get('status')
+    if status!='Success':
+        return ''
+    
+    room = '-'.join(data.get('event_name').split(' ')) + '-' + str(data['otp'])
+    username = data.get("username") #add this to payLoad (clientside)
+    join_room(room)
+    mymessage = {
+        'username':data.get('username'),
+        'RoomName':room
+    }
+
+    emit('join_room', json.dumps(mymessage), room=room)
+
+
+
+@socketio.on('admin_stats', namespace='/admin_namespace')
+def made_for_admin():
+
+    """
+    To tell flask-SocketIO that there exsists a namespace named admin_namespace
+    then we can use it in functions written below. Or else error will be thrown.
+    """
+    #admin_stats is mapped with admin_listen
+    pass
+
+
+
+@socketio.on('attendence_request', namespace='/attendence_namespace')
+def take_attendence_from_user(json_):
+    
+    #attendence_request is mapped with attendence_result at now
+    """
+    This will mark attendence if valid or return failure reason
+    at the namespace 'attendence_namespace' and this is done privately for
+    the user.
+    It also broadcasts the attendence result to
+    """
+
+    otp_ = json_['otp']
+    token_ = json_['token']
+
+
+    user_detail = user_info(token_)
+    if user_detail != 'AuthFail':
```

Comment:
Use status code here rather than string matching the status messages.

Reply:
AuthFail is coming from a function which will remain on server side throughout.
Although it's implementation can also be done in dict form but now it's used at many places already and as I say that it's a return from MyFunction function user_info so it's used as a flag here.